src/modules/0.0.18/aws_memory_db/mappers/memory_db_cluster.ts

```diff
@@ -0,0 +1,266 @@
+          const data = await clnt.describeClusters(cmdInput);
+          for (const cluster of data?.Clusters ?? []) {
+            if (cluster.Status !== readyStatus) {
+              return { state: WaiterState.RETRY };
+            }
+          }
+          return { state: WaiterState.SUCCESS };
+        } catch (e: any) {
+          if (e.name === 'ClusterNotFoundFault') {
+            return { state: WaiterState.SUCCESS };
+          }
+          throw e;
+        }
+      },
+    );
+  };
+
+  cloud: Crud2<MemoryDBCluster> = new Crud2({
+    create: async (es: MemoryDBCluster[], ctx: Context) => {
+      const client = (await ctx.getAwsClient()) as AWS;
+      const out = [];
+      for (const e of es) {
+        // Check if subnet group already exists
+        if (!e.subnetGroup.arn) throw new Error('Subnet group need to be created first');
+        // Now create the cluster
+        const input: CreateClusterCommandInput = {
+          ACLName: 'open-access',
+          ClusterName: e.clusterName,
+          NodeType: e.nodeType,
+          Description: e.description,
+          Port: e.port,
+          SecurityGroupIds: e.securityGroups?.map(sg => sg.groupId ?? '') ?? [],
+          SubnetGroupName: e.subnetGroup.subnetGroupName,
+        };
+        // todo: add tags
+        const newClusterName = await this.createCluster(client.memoryDBClient, input);
+        await this.waitClusterUntil(client.memoryDBClient, newClusterName ?? '', 'available');
+        // Re-get the inserted record to get all of the relevant records we care about
+        const newObject = await this.getCluster(client.memoryDBClient, newClusterName);
+        if (!newObject) continue;
+        // We map this into the same kind of entity as `obj`
+        const newEntity = await this.memoryDBClusterMapper(newObject, ctx);
+        if (!newEntity) continue;
+        // Save the record back into the database to get the new fields updated
+        newEntity.id = e.id;
+        await this.module.memoryDBCluster.db.update(newEntity, ctx);
+        out.push(newEntity);
+      }
+      return out;
+    },
+    read: async (ctx: Context, id?: string) => {
+      const client = (await ctx.getAwsClient()) as AWS;
+      if (id) {
+        const rawCluster = await this.getCluster(client.memoryDBClient, id);
+        if (!rawCluster) return;
+        return this.memoryDBClusterMapper(rawCluster, ctx);
+      } else {
+        const rawClusters = (await this.getClusters(client.memoryDBClient)) ?? [];
+        const out = [];
+        for (const cl of rawClusters) {
+          const outCl = await this.memoryDBClusterMapper(cl, ctx);
+          if (outCl) out.push(outCl);
+        }
+        return out;
+      }
+    },
+    updateOrReplace: (prev: MemoryDBCluster, next: MemoryDBCluster) => {
+      if (
+        !Object.is(prev?.port, next?.port) ||
+        !Object.is(prev?.subnetGroup?.subnetGroupName, next?.subnetGroup?.subnetGroupName)
+      )
+        return 'replace';
```

Comment:
Nit: If the body of the `if` statement is on another line, can we have `{}` surrounding it?

One-liner `if` statements without the brackets is pretty clear, but multi-line gets difficult to parse visually.